src/System.Slices/src/System/Span.cs

```diff
@@ -87,6 +87,8 @@ public Span(T[] array, int start, int length)
         {
             Contract.Requires(array != null);
             Contract.RequiresInInclusiveRange(start, array.Length);
+            Contract.RequiresNonNegative(length);
+            Contract.RequiresInInclusiveRange(start + length, array.Length);
             if (start < array.Length)
```

Comment:
All of them are needed:
`Contract.RequiresInInclusiveRange(start, array.Length);` - case where start can be negative or completely outside of range.
`Contract.RequiresNonNegative(length);` - case where length is negative but both calculated start and end are within acceptable range
`Contract.RequiresInInclusiveRange` - start withing accepotable range, length positive but too big to fit in the array.

What could work is changing first range check with checking if start is positive as the start + length would detect the range problem - I personally prefer to keep them separate as I have a feeling we will end up with different exception message for both of them after design reviews (hopefully compiler will detect this simple double range check anyway and optimize)